protocols/grpc/agent.proto

```diff
@@ -195,15 +194,40 @@ message RouteRequest {
 	Route route = 1;
 }
 
-message OnlineCPUMemRequest {
-}
-
-message Storage {
-	string driver = 1;           // empty in most cases. it will support "drbd", "bcache" ...
-	string source = 2;           // "/dev/sdb", "/dev/disk/by-scsi/xxxx", "none",...
-	string fstype = 3;           // "xfs", "ext4" etc. for block dev, or "9p" for shared fs, or  "tmpfs" for shared /dev/shm for all containers ...
-	repeated string options = 4; // options for the storage device, might required by some special devices.
-	string mount_point = 5;      // mount_point is only visible by VM, not by containers. This mount point can be used on oci.Mount.Source as "/Stroage/mount/point/{rootfs|_data}".
+message Device {
+	string device_id = 1;
+	string driver = 2;
+	map<string, string> driver_options = 3;
+	// device could be "/dev/sdb", "/dev/disk/by-scsi/xxxx", "none", "share_tag" ...
+	string device = 4;
+
+	// When the device is used for storage and requested to be mounted by agent, the fstype and mount_point should be set.
+	// This is usefull for the cases that libcontainer can't handle:
+	// 1) special drivers are involved
+	// 2) libcontainer can't recognize the device
+	// 3) multiple containers need to share the device which multiple-mount is not workable such as a shared tmpfs.
+	// 4) the source for the OCI container's mount is not the root directory of the device, such as
+	//    /mount/point/of/block-device/rootfs (docker image on block)
+	//    /mount/point/of/block-device/_data (docker volume on block)
+	// 5) other cases ...
+	// But in the case that libcontainer can handle, it is recommended to set the config in OCI Container Spec.
+
+	// "xfs", "ext4" etc. for block dev, or "9p" for shared fs, or  "tmpfs" for shared /dev/shm for all containers ...
+	string fstype = 5;
+	// options for the storage device.
+	repeated string mount_options = 6;
+	// mount_point is only visible by VM, not by containers.
+	// This mount point can be used on oci.Spec.Mount.Source such as "/Stroage/mount/point/{rootfs|_data}" (chosen by runtime).
+	string mount_point = 7;
```

Comment:
As @bergwolf suggested, could we put device specific fields into their own structure? Something like:

```
message Storage {
       string fstype = 1;
       repeated string mount_options = 2;
       string mount_point = 3;
}

message Device {
       string device_id = 1;
       string driver = 2;
       map<string, string> driver_options = 3;
       // device could be "/dev/sdb", "/dev/disk/by-scsi/xxxx", "none", "share_tag" ...
       string device = 4
       oneof properties {
              Storage storage = 5;
       }
}
```